Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -896,6 +903,57 @@ public void onStopTrackingTouch(SeekBar seekBar)
     builder.show();
   }
 
+  private void adjustHapticFeedback() {
+    LayoutInflater inflater = LayoutInflater.from(this);
+    View view = inflater.inflate(R.layout.dialog_seekbar, null);
+    int defaultHapticLevel = 0;
+
+    final SeekBar seekbar = view.findViewById(R.id.seekbar);
+    final TextView value = view.findViewById(R.id.text_value);
+    final TextView units = view.findViewById(R.id.text_units);
+
+    seekbar.setMax(255);
+    seekbar.setMin(0);
+    seekbar.setProgress(mPreferences.getInt("hapticFeedbackLevel", defaultHapticLevel));
+    seekbar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+      public void onStartTrackingTouch(SeekBar seekBar) {
+      }
+
+      public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        value.setText(String.valueOf(Math.round(progress / 255.0 * 100)));
+      }
+
+      public void onStopTrackingTouch(SeekBar seekBar) {
+        setHapticFeedbackLevel(seekbar.getProgress());
+      }
+    });
+
+    value.setText(String.valueOf(Math.round(seekbar.getProgress() / 255.0 * 100)));
+    units.setText("%");
+
+    AlertDialog.Builder builder = new AlertDialog.Builder(this);
+    builder.setTitle(R.string.emulation_haptic_feedback);
+    builder.setView(view);
+    final int previousProgress = seekbar.getProgress();
+    builder.setNegativeButton(android.R.string.cancel, (dialogInterface, i) -> {
+      setHapticFeedbackLevel(previousProgress);
+    });
+    builder.setPositiveButton(android.R.string.ok, (dialogInterface, i) -> {
+      setHapticFeedbackLevel(seekbar.getProgress());
+    });
+    builder.setNeutralButton(R.string.emulation_haptic_feedback_reset, (dialogInterface, i) -> {
+      setHapticFeedbackLevel(defaultHapticLevel);
+    });
+
+    builder.create().show();
+  }
+
+  private void setHapticFeedbackLevel(int hapticFeedbackLevel) {
+    SharedPreferences.Editor editor = mPreferences.edit();
+    editor.putInt("hapticFeedbackLevel", hapticFeedbackLevel);
+    editor.apply();
```

Comment:
We're trying to migrate away from using `SharedPreferences`. Please add your new setting here instead: https://github.com/dolphin-emu/dolphin/blob/ffdc8538a162b1ca56e3943c5a78c02358053b2a/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/IntSetting.java#L27-L32